### PR TITLE
ResourceGraph SDK changes for 2021-03-01 API version

### PR DESCRIPTION
**Pull request by fenwickt**
- Generated contracts for 2021-03-01 api-version from resourcegraph.json
- Updated SDK version to 2.1.0
- Added Manual QueryRequest partial class to pereserve the backward compatibility (as per feedback from the previous PR)
- Added tests for new tenant and MG scenarios
- Breaking change - default resultFormatOption is now objectArray instead of table. Our RP still supports 'table' as a default format for older API versions, but it uses 'objectArray' as a default for 2021-03-01

Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/13380
Automated PR in this repo: https://github.com/AzureSDKAutomation/azure-sdk-for-net/pull/3486/files

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -168,6 +168,7 @@
     <PackageReference Update="Microsoft.Azure.Management.EventGrid" Version="4.0.1-preview" />
     <PackageReference Update="Microsoft.Azure.Management.EventHub" Version="2.5.0" />
     <PackageReference Update="Microsoft.Azure.Management.HDInsight" Version="4.1.0-preview" />
+    <PackageReference Update="Microsoft.Azure.Management.ResourceGraph" Version="2.1.0" />
```
Comment (fenwickt) [inline eng/Packages.Data.props:171]: please advice if this change is needed, it was added manually

File: sdk/resourcegraph/Microsoft.Azure.Management.ResourceGraph/src/Manual/QueryRequest.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        /// This constructor is used in the older versions of ResourceGraph
+        /// package and is needed for the backward compatibility.
+        /// </summary>
+        /// <param name="subscriptions">Azure subscriptions against which to
+        /// execute the query.</param>
+        /// <param name="query">The resources query.</param>
+        /// <param name="options">The query evaluation options</param>
+        /// <param name="facets">An array of facet requests to be computed
+        /// against the query result.</param>
+        public QueryRequest(IList<string> subscriptions, string query, QueryRequestOptions options = default(QueryRequestOptions), IList<FacetRequest> facets = default(IList<FacetRequest>))
```
Comment (dramuy) [inline sdk/resourcegraph/Microsoft.Azure.Management.ResourceGraph/src/Manual/QueryRequest.cs:21]: >QueryRequest [](start = 15, length = 12)

guess this constructor is for backward compatibility

**Comment by dramuy** [comment]
> Installing AutoRest version: v2

is this needed to add to repo?

---
Refers to: eng/mgmt/mgmtmetadata/resourcegraph_resource-manager.txt:1 in b96821f. [](commit_id = b96821f0a8a469ed79eb57824c1b87bd5dd9ed59, deletion_comment = False)

**Comment by allenjzhang** [comment]
> Installing AutoRest version: v2

Yes.Thanks.

---
In reply to: [804340490](https://github.com/Azure/azure-sdk-for-net/pull/19536#issuecomment-804340490) [](ancestors = 804340490)

---
Refers to: eng/mgmt/mgmtmetadata/resourcegraph_resource-manager.txt:1 in b96821f. [](commit_id = b96821f0a8a469ed79eb57824c1b87bd5dd9ed59, deletion_comment = False)

File: sdk/resourcegraph/Microsoft.Azure.Management.ResourceGraph/src/Manual/QueryRequest.cs (hunk 2 above)
Comment (YalinLi0312) [reply sdk/resourcegraph/Microsoft.Azure.Management.ResourceGraph/src/Manual/QueryRequest.cs:21]: Yes, but please change the folder name from "Manual" to "Customization"

**Review by YalinLi0312 — CHANGES_REQUESTED** [review]

**Review by YalinLi0312 — APPROVED** [review]

**Pull request merged**